lib/core/time/maps.js

```diff
@@ -203,45 +203,34 @@ exports.MapController = function(options, timeControls) {
         map = options.map;
     function layerAdded(layer) {
         var source, image;
+        var loaded = function(event) {
+            // grab the active loadListener to avoid phantom onloads
+            // when listener is cancelled
+            var currentListener = loadListener;
+            if (currentListener) {
+                currentListener.tileLoaded(event, source);
+            }
+        };
+        var loadstart = function() {
+            // grab the active loadListener to avoid phantom onloads
+            // when listener is cancelled
+            var currentListener = loadListener;
+            if (currentListener) {
+                currentListener.tileQueued(source);
+            }
+        };
         if (layer instanceof ol.layer.Tile && layer.getSource() instanceof ol.source.TileWMS) {
             source = layer.getSource();
-            source.on('tileloadstart', function() {
-                // grab the active loadListener to avoid phantom onloads
-                // when listener is cancelled
-                var currentListener = loadListener;
-                if (currentListener) {
-                    currentListener.tileQueued(source);
-                }
-            });
-            var loaded = function(event) {
-                // grab the active loadListener to avoid phantom onloads
-                // when listener is cancelled
-                var currentListener = loadListener;
-                if (currentListener) {
-                    currentListener.tileLoaded(event, source);
-                }
-            };
+            source.on('tileloadstart', loadstart);
             source.on('tileloadend', loaded);
             // @todo handle onerror and cancel deferred with an example
             // to stop automatic playback
             source.on('tileloaderror', loaded);
         } else if (layer instanceof ol.layer.Image && layer.getSource() instanceof ol.source.ImageWMS) {
             source = layer.getSource();
-            source.setImageLoadFunction((function() {
-                return function(image, src) {
-                    // grab the active loadListener to avoid phantom onloads
-                    // when listener is cancelled
-                    var currentListener = loadListener;
-                    image = image.getImage();
-                    if (currentListener) {
-                        currentListener.tileQueued(source);
-                        image.onload = image.onerror = function(event) {
-                            currentListener.tileLoaded(event, source);
-                        };
-                    }
-                    image.src = src;
-                };
-            })());
+            source.on('imageloadstart', loadstart);
+            source.on('imageloadend', loaded);
+            source.on('imageloaderror', loaded);
```

Comment:
no there is nothing in the event for this, this would need images to be loaded over XHR like @ahocevar does in composer I think